### PR TITLE
perf: simplify swapper asset sorting for faster boot times

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Does the following:
- While market data is loading, the swapper asset list is sorted by name alphabetically
- After market data arrives, sorts assets the same way as in production (user balance, market cap, name)
- Corrects loading state in swapper such that loading state is true while assets are still being loaded
- No longer refetches swapper supported assets based on asset sorting

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

progresses #5975

### CODE REVIEW

**Comment by woodenfurniture** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #6007** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6007?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/6007?utm_source=stack-comment).

File: src/components/MultiHopTrade/hooks/useSupportedAssets.tsx [hunk 1]
```diff
@@ -1,43 +1,47 @@
...
+} from 'state/slices/common-selectors'
+import { selectMarketDataDidLoad } from 'state/slices/selectors'
+import { store, useAppSelector } from 'state/store'
 
 export const useSupportedAssets = () => {
-  const sortedAssets = useAppSelector(selectAssetsSortedByMarketCapUserCurrencyBalanceAndName)
-  const assets = useAppSelector(selectAssets)
-  const wallet = useWallet().state.wallet
-  const isSnapInstalled = useIsSnapInstalled()
+  const marketDataDidLoad = useAppSelector(selectMarketDataDidLoad)
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/hooks/useSupportedAssets.tsx:12]: 🧠

File: src/components/MultiHopTrade/hooks/useSupportedAssets.tsx [hunk 2]
```diff
@@ -1,43 +1,47 @@
...
+  }, [walletSupportedChainIds])
 
-  const { data, isLoading } = useGetSupportedAssetsQuery(queryParams)
+  const { data, isFetching } = useGetSupportedAssetsQuery(queryParams)
 
   const supportedSellAssets = useMemo(() => {
     if (!data) return []
-    return data.supportedSellAssetIds.map(assetId => assets[assetId]).filter(isSome)
-  }, [assets, data])
+    const assetIdsSet = new Set(data.supportedSellAssetIds)
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/hooks/useSupportedAssets.tsx:33]: Speed 🚀

Comment (gomesalexandre) [inline src/components/MultiHopTrade/hooks/useSupportedAssets.tsx:33]: blazingly fast 🧠

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
Tested locally with a full cache clear every test:

- MM (EVM only)

https://github.com/shapeshift/web/assets/17035424/156715e4-9d7c-4058-90c8-585c1c7915b7

- Native


https://github.com/shapeshift/web/assets/17035424/f8eb47d0-1408-4dc0-9374-0db9dd87a4cf

- Native against develop, note how sorting by portfolio balance is now broken 😢 

https://github.com/shapeshift/web/assets/17035424/f0d5937d-e1cc-4ad0-8d2b-4a0b3026261d

Note, switching back and forth from the trade page succesfully sorts by balances/market cap, most likely because of the non-reactivity of https://github.com/shapeshift/web/pull/6007#discussion_r1453308207:


https://github.com/shapeshift/web/assets/17035424/0b28e55a-364c-49d7-88e1-be35db9878a1

**Review by gomesalexandre — APPROVED** [review]
Retested locally, confirmed reactivity rug is fixed 

https://github.com/shapeshift/web/assets/17035424/7d8cfb6e-2626-4517-bcd8-fe344a4d3c63

**Pull request merged**